cli/src/commands/invite/claim.rs

```diff
@@ -69,9 +78,58 @@ pub async fn main(args: Args) -> anyhow::Result<()> {
             let ctx = step4_device(ctx).await?;
             save_device(ctx, save_mode).await
         }
-        AnyClaimRetrievedInfoCtx::ShamirRecovery(_) => Err(anyhow::anyhow!(
-            "Shamir recovery invitation is not supported yet"
-        )),
+        AnyClaimRetrievedInfoCtx::ShamirRecovery(ctx) => {
+            let mut pick_ctx = ctx;
+
+            let mut device_ctx = loop {
```

Comment:
What happen if the user kill the CLI mid-process to selecting the recipients?
Do we have a cleanup to do?